nginx/crowdsec_nginx.conf

```diff
@@ -1,40 +1,126 @@
 lua_package_path '/usr/local/lua/crowdsec/?.lua;;';
 lua_shared_dict crowdsec_cache 50m;
+
 lua_ssl_trusted_certificate /etc/ssl/certs/ca-certificates.crt;
+
+#
+# INITIALIZATION (runs once on Nginx start)
+#
 init_by_lua_block {
-        cs = require "crowdsec"
-        local ok, err = cs.init("/etc/crowdsec/bouncers/crowdsec-nginx-bouncer.conf", "crowdsec-nginx-bouncer/v1.1.3")
-        if ok == nil then
-                ngx.log(ngx.ERR, "[Crowdsec] " .. err)
-                error()
-        end
+    -- ## FIX 1: Use pcall and a global flag for safer initialization ##
+    -- This ensures that if CrowdSec fails to start, the other blocks will not run and cause errors.
+    cs = require "crowdsec"
+    local config_path = "/data/crowdsec/crowdsec.conf"
+    local user_agent  = "crowdsec-npmplus-bouncer/v1.1.3"
+
+    local ok, err = pcall(cs.init, config_path, user_agent)
+    if not ok then
+        ngx.log(ngx.ERR, "[Crowdsec] FATAL: Init failed: " .. tostring(err))
+        _G.CROWDSEC_INIT_FAILED = true
+    else
         ngx.log(ngx.ALERT, "[Crowdsec] Initialisation done")
+        _G.CROWDSEC_INIT_FAILED = false
+    end
 }
 
 map $server_addr $unix {
-	default       0;
-	"~unix:" 1;
+    default 0;
+    "~unix:" 1;
 }
 
+#
+# PER-REQUEST PROCESSING (checks every request against the local cache)
+#
 access_by_lua_block {
-	local cs = require "crowdsec"
-	if ngx.var.unix == "1" then
-		ngx.log(ngx.DEBUG, "[Crowdsec] Unix socket request ignoring...")
-	else
-		cs.Allow(ngx.var.remote_addr)
-	end
+    if _G.CROWDSEC_INIT_FAILED then return end
+
+    local cs = require "crowdsec"
+    if ngx.var.unix == "1" then return end
+
+    local ok, err = pcall(cs.Allow, ngx.var.remote_addr)
+    if not ok then
+        ngx.log(ngx.ERR, "[Crowdsec] Allow() failed unexpectedly: " .. tostring(err))
+    end
 }
 
+#
+# WORKER INITIALIZATION (runs the crash-proof supervisor)
+#
 init_worker_by_lua_block {
-        cs = require "crowdsec"
-        local mode = cs.get_mode()
-        if string.lower(mode) == "stream" then
-           ngx.log(ngx.INFO, "Initializing stream mode for worker " .. tostring(ngx.worker.id()))
-           cs.SetupStream()
+    if _G.CROWDSEC_INIT_FAILED then
+        ngx.log(ngx.WARN, "[Crowdsec] Bouncer is disabled due to initialization failure.")
```

Comment:
In a default configuration, nginx does not write warning to the error log, so users are very unlikely to see this message.